website/static/js/pages/project-settings-page.js

```diff
@@ -128,25 +128,70 @@ $(document).ready(function() {
             var $elm = $(elm);
             formData[$elm.attr('name')] = $elm.is(':checked');
         });
+
+        var unchecked = checkedOnLoad.filter('#selectAddonsForm input:not(:checked)');
         var msgElm = $(this).find('.addon-settings-message');
-        $.ajax({
-            url: ctx.node.urls.api + 'settings/addons/',
-            data: JSON.stringify(formData),
-            type: 'POST',
-            contentType: 'application/json',
-            dataType: 'json',
-            success: function() {
-                msgElm.text('Settings updated').fadeIn();
-                checkedOnLoad = $('#selectAddonsForm input:checked');
-                uncheckedOnLoad = $('#selectAddonsForm input:not(:checked)');
-                if($osf.isSafari()){
-                    //Safari can't update jquery style change before reloading. So delay is applied here
-                    setTimeout(function(){window.location.reload();}, 100);
-                } else {
-                    window.location.reload();
-                }
+        
+        var submit = function() {
+            var request = $osf.postJSON(ctx.node.urls.api + 'settings/addons/', formData);
+            return request;
+            
+        };
+        function successfulAddonUpdate(){
+            msgElm.text('Settings updated').fadeIn();
+            checkedOnLoad = $('#selectAddonsForm input:checked');
+            uncheckedOnLoad = $('#selectAddonsForm input:not(:checked)');
+            if($osf.isSafari()){
+                        //Safari can't update jquery style change before reloading. So delay is applied here
+                        setTimeout(function(){window.location.reload();}, 100);
+            } else {
+                        window.location.reload();
             }
-        });
+        }
+        function failedAddonUpdate(){
+            var msg = 'Sorry, we had trouble saving your settings. If this persists please contact <a href="mailto: support@osf.io">support@osf.io</a>';
+            bootbox.alert({
+                title: 'Request failed',
+                message: msg,
```

Comment:
Add a space after the `:` for this and the several lines below. (minor style point)